ql/pricingengines/barrier/analyticbarrierengine.cpp

```diff
@@ -178,13 +178,17 @@ namespace QuantLib {
         return (1 + mu()) * stdDeviation();
     }
 
+    DiscountFactor AnalyticBarrierEngine::costOfCarry() const {
+        return  dividendDiscount() / riskFreeDiscount();
```

Comment:
The cost of carry is the difference between risk-free rate and dividend yield.  This is the corresponding discount factor, not the cost of carry itself.